instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ManifestResourceProvider.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.resources;
+
+import static java.util.logging.Level.WARNING;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.semconv.ResourceAttributes;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.jar.Manifest;
+import java.util.logging.Logger;
+
+/**
+ * A {@link ResourceProvider} that will attempt to detect the <code>service.name</code> and <code>
+ * service.version</code> from META-INF/MANIFEST.MF.
+ */
+@AutoService(ResourceProvider.class)
+public final class ManifestResourceProvider extends AttributeResourceProvider<Manifest> {
+
+  private static final Logger logger = Logger.getLogger(ManifestResourceProvider.class.getName());
+
+  @SuppressWarnings("unused") // SPI
+  public ManifestResourceProvider() {
+    this(new JarPathFinder(), ManifestResourceProvider::readManifest);
```

Comment:
both this class and `JarServiceNameDetector` attempt to find the main jar, you could find it only once and use the result for both

Reply:
we already do this: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10621/files#diff-9274d424baf0a4a58f5c616a81cf4ec659fce87b4e13e6f423414fe0e2a0602cR52-R57

Reply:
Didn't expect that. Instead of passing around `JarPathFinder` I would have passed around `Supplier<Optional<Path>>` I would have create a separate class `MainJarPathHolder` that has `private static final Optional<Path> jarPath = new JarPathFinder().getJarPath();` and used the jar path from the holder in both detectors.

Reply:
good idea - let me try that

Reply:
I've done it a bit differently now, because the finder needs to be passed around in unit tests